ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/websocket/subscription/blockheaders/NewBlockHeadersSubscriptionServiceTest.java

```diff
@@ -63,76 +73,116 @@
   @Captor ArgumentCaptor<JsonRpcResult> responseCaptor;
 
   @Mock private SubscriptionManager subscriptionManager;
-  @Mock private BlockchainQueries blockchainQueries;
 
   private final BlockHeaderTestFixture blockHeaderTestFixture = new BlockHeaderTestFixture();
   private final TransactionTestFixture txTestFixture = new TransactionTestFixture();
   private final BlockHeader blockHeader = blockHeaderTestFixture.buildHeader();
   private final BlockResultFactory blockResultFactory = new BlockResultFactory();
+  private final BlockDataGenerator gen = new BlockDataGenerator();
+  private final BlockchainStorage blockchainStorage =
+      new KeyValueStoragePrefixedKeyBlockchainStorage(
+          new InMemoryKeyValueStorage(), new MainnetBlockHeaderFunctions());
+  private final Block genesisBlock = gen.genesisBlock();
+  private final DefaultBlockchain blockchain =
+      (DefaultBlockchain)
+          DefaultBlockchain.createMutable(genesisBlock, blockchainStorage, new NoOpMetricsSystem());
+
+  @Spy
+  private BlockchainQueries blockchainQueriesSpy =
+      Mockito.spy(new BlockchainQueries(blockchain, createInMemoryWorldStateArchive()));
 
   @Before
   public void before() {
     newBlockHeadersSubscriptionService =
-        new NewBlockHeadersSubscriptionService(subscriptionManager, blockchainQueries);
+        new NewBlockHeadersSubscriptionService(subscriptionManager, blockchainQueriesSpy);
   }
 
   @Test
   public void shouldSendMessageWhenBlockAddedOnCanonicalChain() {
     final NewBlockHeadersSubscription subscription = createSubscription(false);
     mockSubscriptionManagerNotifyMethod(subscription);
-    final BlockResult expectedNewBlock = expectedBlockWithTransactions(Collections.emptyList());
 
-    simulateAddingBlockOnCanonicalChain();
+    final Block testBlock = appendBlockWithParent(blockchain, blockchain.getChainHeadBlock());
+    simulateAddingBlockOnCanonicalChain(testBlock);
+    final BlockResult expectedNewBlock =
+        blockResultFactory.transactionHash(
+            blockchainQueriesSpy.blockByHashWithTxHashes(testBlock.getHash()).orElse(null));
 
     verify(subscriptionManager)
         .sendMessage(subscriptionIdCaptor.capture(), responseCaptor.capture());
     assertThat(subscriptionIdCaptor.getValue()).isEqualTo(subscription.getSubscriptionId());
-    assertThat(responseCaptor.getValue())
-        .isEqualToComparingFieldByFieldRecursively(expectedNewBlock);
+    assertThat(responseCaptor.getValue()).usingRecursiveComparison().isEqualTo(expectedNewBlock);
   }
 
   @Test
   public void shouldNotSendMessageWhenBlockAddedIsNotOnCanonicalChain() {
     simulateAddingBlockOnNonCanonicalChain();
 
-    verifyZeroInteractions(subscriptionManager);
+    verifyNoMoreInteractions(subscriptionManager);
```

Comment:
Why did you choose `VerifyNoMoreInteractions` over `VerifyNoInteractions`?

Reply:
I misread the javadoc where it says VerifyNoMoreInteractions does the same thing. Will update to VerifyNoInteractions.